python/ray/tune/utils/object_cache.py

```diff
@@ -0,0 +1,168 @@
+from collections import defaultdict, Counter
+
+from typing import Dict, Generator, List, Optional, TypeVar
+
+# Grouping key - must be hashable
+T = TypeVar("T")
+# Objects to cache
+U = TypeVar("U")
+
+
+class _ObjectCache:
+    """Cache up to some maximum count given a grouping key.
+
+    This object cache can e.g. be used to cache Ray Tune trainable actors
+    given their resource requirements (reuse_actors=True).
+
+    If the max number of cached objects for a grouping key is reached,
+    no more objects for this group will be cached.
+
+    However, if eager caching is enabled, one object (globally across all grouping
+    keys) may be cached, even if the max number of objects is 0. This is to
+    allow to cache an object if the max number of objects of this key
+    will increase shortly after (as is the case e.g. in the Ray Tune control
+    loop).
+
+    Args:
+        eager_caching: If True, one object (globally) may be cached even if
```

Comment:
`eager_caching` gives me a different impression just by reading the name of it. 
This also feels so catered towards Tune's use case (although I only vaguely understand why our control loop may need one more actor. 
If we cannot make this class's API generic, maybe we can rename this `eager_caching` arg to be more specific? Like `cache_one_more`?

Reply:
Renamed to `may_keep_one`